### ISSUE TITLE
Must use splat syntax to access X,  because it has "count" set

### ISSUE THREAD

**Issue by slmingol** [cloudposse/terraform-aws-cloudwatch-flow-logs#6]
With the following module setup I'm getting a failure:
```
Warning: output "log_group_arn": must use splat syntax to access aws_cloudwatch_log_group.default attribute "arn", because it has "count" set; use aws_cloudwatch_log_group.default.*.arn to obtain a list of the attributes across all instances



Warning: output "vpc_flow_id": must use splat syntax to access aws_flow_log.vpc attribute "id", because it has "count" set; use aws_flow_log.vpc.*.id to obtain a list of the attributes across all instances



Warning: output "kinesis_id": must use splat syntax to access aws_kinesis_stream.default attribute "id", because it has "count" set; use aws_kinesis_stream.default.*.id to obtain a list of the attributes across all instances



Warning: output "kinesis_name": must use splat syntax to access aws_kinesis_stream.default attribute "name", because it has "count" set; use aws_kinesis_stream.default.*.name to obtain a list of the attributes across all instances



Warning: output "kinesis_shard_count": must use splat syntax to access aws_kinesis_stream.default attribute "shard_count", because it has "count" set; use aws_kinesis_stream.default.*.shard_count to obtain a list of the attributes across all instances



Warning: output "kinesis_arn": must use splat syntax to access aws_kinesis_stream.default attribute "arn", because it has "count" set; use aws_kinesis_stream.default.*.arn to obtain a list of the attributes across all instances
```

If I follow the suggestions for the above it then seems to work.

Here's my flow_logs module:
```
module "flow_logs" {
  source    									= "git::https://github.com/cloudposse/terraform-aws-cloudwatch-flow-logs.git?ref=master"
  vpc_id    									= "${var.vpc_id}"
  namespace 									= "${var.ns}"
  stage     									= "${var.env}"
}
```

**Comment by ChrisMcKee** [comment]
Also "log_group_name": [DEPRECATED] use 'log_destination' argument instead

**Comment by nema-sigsci** [comment]
Any update on this? This is affecting us as well.

**Comment by aknysh** [comment]
@ChrisMcKee @nema-sigsci 
that was addressed in https://github.com/cloudposse/terraform-aws-cloudwatch-flow-logs/pull/7, but waiting on @slmingol  to address the comments.
If he could not get to it today, we'll open another PR with the fix tomorrow.

BTW, we should use `splat+join` pattern for all resources with `counts`.

So instead of
value       = "${aws_kinesis_stream.default.*.name}"

we should use
value       = "${join("", aws_kinesis_stream.default.*.name)}"

since when `enabled=false`, it should get an empty string (otherwise it could fail in some cases)

**Comment by aknysh** [comment]
@nema-sigsci please ping me tomorrow so I don't forget to get to it

**Comment by Prophecy67** [comment]
> @nema-sigsci please ping me tomorrow so I don't forget to get to it

Ping :)
